src/CameraOnvif.cpp

```diff
@@ -0,0 +1,408 @@
+#include "CameraOnvif.hpp"
+
+#include <gsoap/plugin/wsseapi.h>
+#include <gsoap/plugin/wsddapi.h>
+#include "onvif/soapDeviceBindingProxy.h"
+#include "onvif/soapImagingBindingProxy.h"
+#include "onvif/soapMediaBindingProxy.h"
+#include "onvif/soapPTZBindingProxy.h"
+#include "onvif/soapPullPointSubscriptionBindingProxy.h"
+#include "onvif/soapRemoteDiscoveryBindingProxy.h"
+#include "wsdd.nsmap"
+
+#include <iostream>
+
+using namespace std;
+using namespace camera_onvif;
+
+struct CameraOnvif::Private {
+    struct soap *soap = soap_new1(SOAP_XML_STRICT | SOAP_XML_CANONICAL | SOAP_C_UTFSTRING);
+    MediaBindingProxy *proxy_media;
+    ImagingBindingProxy *proxy_image;
+    _trt__GetVideoEncoderConfigurationOptionsResponse *video_options;
+}defined;
+
+CameraOnvif::CameraOnvif(/* args */) {
+    user = "admin";
+    pass = "camera01";
+    uri = "http://10.20.0.188/onvif/device_service";
+
+    this->init();
+}
+
+CameraOnvif::CameraOnvif(string user, string pass, string ip){
+    this->user = user;
+    this->pass = pass;
+    this->uri = "http://" + ip + "/onvif/device_service";
+
+    this->init();
+}
+
+CameraOnvif::CameraOnvif(string user, string pass, string ip, int width, int height){
+    this->user = user;
+    this->pass = pass;
+    this->uri = "http://" + ip + "/onvif/device_service";
+
+    this->init();
+    this->setResolution(width, height);
+}
+
+void CameraOnvif::init(){
+    m_private = &defined;
```

Comment:
????

This is a global structure. What happens if I have more than one onvif instance in the same process. As, for instance, connect to three cameras at the same time ?